lib/ukcpio/cpio.c

```diff
@@ -201,12 +203,35 @@ extract_dir(const char *path, mode_t mode)
 	int r;
 
 	uk_pr_info("Creating directory %s\n", path);
-	if ((r = uk_syscall_r_mkdir(path, mode))) {
-		uk_pr_err("%s: Failed to create directory: %s (%d)\n",
-			  path, strerror(-r), -r);
-		return -UKCPIO_MKDIR_FAILED;
+	r = uk_syscall_r_mkdir(path, mode);
+	if (r == -EEXIST) {
```

Comment:
Would it make sense to have `unlikely`'s here and in the following error if cases?

Reply:
As the other checks don't use unlikely, I would aim for consistency first and not add it.

Reply:
That does not mean that it is good. But ok fine. We should at some point have a look at most source files and add `unlikely`'s on error cases where it makes sense (e.g. it makes sense not to use `unlikely` if `newfstatat` is run on an inexistent file, as it is common for apps to test whether a file exists or not through such syscalls but it makes sense to add `unlikely`'s on initrd extraction since it is done by us and if a function fails here then there is not much point in continuing booting).

Reply:
@mogasergiu , fully agree this should be added. My concern is that adding it **now, here** will cause confusion. Why here and not elsewhere?

We would need to be consistent and add it to an entire file / compilation unit.

To keep track of this, I created issue #1233 for this.